### ISSUE TITLE
jquery.a11y - improve combination of style elements and anchor tags

### ISSUE THREAD

**Issue by danielstorey** [adaptlearning/adapt_framework#1727]
Steps to replicate:
Create a text component whose body contains formatted text e.g. `<strong>` elements,  as well at least one link  (`<a>` element)
Turn accessibility on

Current behaviour: because a11y has detected a non-style element (`<a>`) it now creates separate tabbable elements for all style elements

Desired behaviour: links should be tabbable and all other text should be grouped together in a single tabbable block

**Comment by oliverfoster** [comment]
We may have come across this before... Does the outcome change if you wrap the whole thing in a ``<p>`` ?

**Comment by danielstorey** [comment]
Yeah, still the same. I understand the challenge involved in trying to achieve the above, especially if the `<a>` is wrapped in a `<strong>` with some other text. Maybe there's a middle ground if the desired behaviour as I've described is not achievable. I'll have a think about possible solutions.

**Comment by chris-steele** [comment]
I agree it's worth revisiting this because of the real impact, particularly for screen reader users, if the read order is punctuated unnaturally. I'll have a think about this and see if I can describe the behaviours we want with this functionality and once we're agreed we can try and implement (good bit of BDD!). This said, there's always the edge cases we miss so let's also implement a feature that allows us to state explicitly in the data how we want the text to be read: e.g. wrapping the blocks of content we want to be a single tabbable unit in a class that will override the default functionality. This will save users having to wait for fixes/updates or hack the code.

**Comment by oliverfoster** [comment]
could you try this extension https://github.com/cgkineo/adapt-a11yText
see if that helps the reading situation? @danielstorey

**Comment by moloko** [comment]
done in 64f635a076cb08cff312e208075f59e17ba9e835
